src/tests/ftest/control/dmg_telemetry_io_basic.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from avocado.core.exceptions import TestFail
+
+from ior_test_base import IorTestBase
+from telemetry_test_base import TestWithTelemetry
+
+
+class TestWithTelemetryIOBasic(IorTestBase,TestWithTelemetry):
+    # pylint: disable=too-many-ancestors
+    """Test telemetry engine io basic metrics.
+
+    :avocado: recursive
+    """
+
+    def verify_io_test_metrics(self, io_test_metrics, metrics_data, threshold):
+        """ Verify telemetry io metrics from metrics_data.
+
+        Args:
+            io_test_metrics (list): list of telemetry io metrics.
+            metrics_data (dict): a dictionary of host keys linked to a
+                                 list of io metric names.
+            threshold (int): test io metrics threshold.
+
+        """
+        for host in self.hostlist_servers:
+            self.log.info("==Host: %s", host)
+            for name in io_test_metrics:
+                self.log.info("  --telemetry metric: %s", name)
+                for i, _ in enumerate(metrics_data):
+                    m_data = metrics_data[i]
+                    if i == 0:
+                        self.log.info("   Initial    : %s", m_data[host][name])
+                    else:
+                        self.log.info("   testloop %s: %s", i,
+                                      m_data[host][name])
```

Comment:
Python dictionary keys are not guaranteed to be in order so if `metrics_data` had the following keys `0, 1, 2, 3` then `for index, key in enumerate(metrics_data)` could yield: `(index=0, key=2), (index=1, key=0), (index=2, key=3), (index=3, key=1)`.  To avoid this confusion I'd recommend:
```suggestion
                for key sorted(metrics_data):
                    self.log.info(
                        "  %12s: %s",
                        "Initial" if key == 0 else "Test Loop {}".format(key),
                        metrics_data[key][host][name])
```

Reply:
Please see the update.